components/fonts/system_font_service.rs

```diff
@@ -90,80 +89,74 @@ struct ResolvedGenericFontFamilies {
+
+        SystemFontServiceProxySender(sender)
+    }
+
     #[tracing::instrument(skip(self), fields(servo_profiling = true))]
     fn run(&mut self) {
         loop {
             let msg = self.port.recv().unwrap();
 
             match msg {
-                Command::GetFontTemplates(descriptor_to_match, font_family, result) => {
-                    let span = span!(
-                        Level::TRACE,
-                        "Command::GetFontTemplates",
-                        servo_profiling = true
-                    );
+                Command::GetFontTemplates(font_descriptor, font_family, result_sender) => {
+                    let span = span!(Level::TRACE, "GetFontTemplates", servo_profiling = true);
                     let _span = span.enter();
-                    let templates =
-                        self.find_font_templates(descriptor_to_match.as_ref(), &font_family);
-                    debug!("Found templates for descriptor {descriptor_to_match:?}: ");
-                    debug!("  {templates:?}");
-
-                    let (serialized_templates, senders): (
-                        Vec<SerializedFontTemplate>,
-                        Vec<(FontTemplateRef, IpcBytesSender)>,
-                    ) = templates
-                        .into_iter()
-                        .map(|template| {
-                            let (bytes_sender, bytes_receiver) =
-                                ipc::bytes_channel().expect("failed to create IPC channel");
-                            (
-                                SerializedFontTemplate {
-                                    identifier: template.identifier().clone(),
-                                    descriptor: template.descriptor().clone(),
-                                    bytes_receiver,
-                                },
-                                (template.clone(), bytes_sender),
-                            )
-                        })
-                        .unzip();
-
-                    let _ = result.send(serialized_templates);
-
-                    // NB: This will load the font into memory if it hasn't been loaded already.
-                    for (font_template, bytes_sender) in senders.iter() {
-                        let identifier = font_template.identifier();
-                        let data = self
-                            .font_data
-                            .entry(identifier)
-                            .or_insert_with(|| font_template.data());
-                        let span = span!(
-                            Level::TRACE,
-                            "GetFontTemplates send",
-                            servo_profiling = true
-                        );
-                        let _span = span.enter();
-                        let _ = bytes_sender.send(data);
-                    }
+                    let _ =
+                        result_sender.send(self.get_font_templates(font_descriptor, font_family));
```

Comment:
By the way, this, combined with [this recv](https://github.com/servo/servo/pull/33530/files#diff-5345e8d47974e67c3ddf58abd80db3fb9dd8bd5b5fb948de9d3a23bd3904ea62L500), is a potential source of deadlock, because of https://github.com/servo/servo/blob/dbd1666b17b22130bcce75e6255f376e6fbe256c/components/constellation/constellation.rs#L75

Reply:
Actually, since the other thread is blocking on the receive, and hence always ready to receive whatever is sent here, it should be ok.